megalinter/utils.py

```diff
@@ -87,11 +87,19 @@ def filter_files(
         else None
     )
 
+    # if each file is check against every ignored_files (it can contain all the files), it's a O(n²) filtering
+    # to reduce the execution time and complexity ignored_files is split
+    ignored_patterns = list(filter(lambda x: '*' in x, ignored_files or []))
+    ignored_fileset = set(ignored_files or [])
+
     # Filter all files to keep only the ones matching with the current linter
 
     for file in all_files:
 
-        if ignored_files and len([n for n in ignored_files if fnmatch(n, file)]) > 0:
+        if file in ignored_fileset:
+            continue
+
+        if ignored_patterns and any(fnmatch(file, pattern) for pattern in ignored_patterns):
```

Comment:
If I'm correct that directories and patterns are no longer a thing with ls-files, you can simplify and remove the pattern matching. fnmatch is still probably a good idea even for file names, in case there's issues with case insensitive OSes.

Reply:
I keep the `fnmatch` to be compatible with ignored_files created manually or with glob pattern (eg if someone in the future, will change the output of `list_git_ignored_files` to return pattern without need to also update this part, or if this function is used with something else than the output of `list_git_ignored_files`)
I agree I keep compatibility with stuff that could be removed.

@nvuillam , what do you prefer, should I remove the "pattern" support or keep it ?

Reply:
Let's keep it like that for now , ascending compatibility is nice :)